include/mqtt/subscribe_options.hpp

```diff
@@ -35,7 +35,7 @@ enum class qos : std::uint8_t
     exactly_once = 0b00000010,
 };
 
-struct subscribe_options
+struct subscribe_options final
 {
     constexpr subscribe_options(void) = delete;
```

Comment:
Could you update coding style as follows? 

```cpp
struct subscribe_options final {	
```

See https://github.com/redboltz/mqtt_cpp/wiki/Coding-Rules